table-webview/src/main.ts

```diff
@@ -0,0 +1,128 @@
+            const action = message.data.action;
+            if (AddRowAction.isThisAction(action)) {
+                this.handleAddRow(action as AddRowAction);
+            } else if (UpdateCellAction.isThisAction(action)) {
+                this.handleUpdateCell(action as UpdateCellAction);
+            } else if (ResetTableAction.isThisAction(action)) {
+                this.handleResetTable();
+            } else if (AddRowListenerAction.isThisAction(action)) {
+                this.addRowListener()
+            }
+        } else {
+            console.log("Message not supported: " + message);
+        }
+    }
+
+    protected addRowListener() {
+        document.addEventListener('click', event => {
+            const action = rowSelection(event);
+            if (action) {
+                vscode.postMessage({ action: action });
+            }
+        });
+    }
+
+    /**
+     * Initializes the webview with a header and a placeholder for the templates.
+     * @param identifier The identifier of the element that should contain the webview.
+     */
+    protected initHtml(identifier: string, headers: string[]): void {
+        this.identifier = identifier;
+        this.headers = headers;
+        const containerDiv = document.getElementById(identifier + '_container');
+        if (containerDiv) {
+            const tablePlaceholder = document.createElement("table");
+            containerDiv.appendChild(tablePlaceholder);
+            const table = createTable(identifier, headers);
+            patch(tablePlaceholder, table);
+        }
+    }
+
+    protected handleAddRow(action: AddRowAction) {
+        const table = document.getElementById(this.identifier + '_table');
+        if (table) {
+            const rowPlaceholder = document.createElement("tr");
+            table.appendChild(rowPlaceholder);
+            const row = createRow(action.rowId, action.values);
+            patch(rowPlaceholder, row);
+        }
+    }
+
+    protected handleUpdateCell(action: UpdateCellAction) {
+        const row = document.getElementById(action.rowId);
+        const index = this.headers.indexOf(action.columnId);
+        const newCell = createCell(action.value);
+        if (index < row.children.length) {
+            patch(row.children[index], newCell);
+        } else {
+            for (let i = row.children.length; i < index; i++) {
+                const cell = document.createElement("td");
+                row.appendChild(cell);
+            }
+            const cellPlaceholder = document.createElement("td");
+            row.appendChild(cellPlaceholder);
+            patch(cellPlaceholder, newCell);
+        }
+    }
+
+    protected handleResetTable() {
+        const table = document.getElementById(this.identifier + '_table');
+        if (table) {
+            const newTable = createTable(this.identifier, this.headers);
+            patch(table, newTable);
+        }
+    }
+
+}
+
+new Starter();
```

Comment:
why export Starter and call it here anyway?

Reply:
From what I understood this should be a generic webview that can be reused for any view. Then this generic code should not be started here, but extended/used and started by the views using this table view.